arrow-libs/fx/arrow-fx-resilience/src/commonMain/kotlin/arrow/fx/resilience/CircuitBreaker.kt

```diff
@@ -724,11 +726,19 @@ private constructor(
 
       override fun trackFailure(failureAt: TimeMark): OpeningStrategy =
         if (failures.size < maxFailures + 1) copy(failures = failures + failureAt)
-        else copy(failures = failures.drop(1) + failureAt)
+        else copy(failures = failures.slide(failureAt))
+
+      private fun List<TimeMark>.slide(timeMark: TimeMark): List<TimeMark> =
+        buildList(size) {
+          forEachIndexed { index, mark ->
+            if (index == 1) Unit else add(mark)
```

Comment:
This was my bad, whoops. `drop(1)` removes `index == 0`.

```suggestion
            if (index == 0) Unit else add(mark)
```